OVERVIEW.md

```diff
@@ -0,0 +1,483 @@
+# Pseudocode
+
+The following attempts to capture the general logic of the Automation OCR2 
+implementation using pseudocode. It is divided first by the OCR2 interface 
+followed by descriptions of secondary services.
+
+## How an Observation is Constructed
+
+An automation OCR2 observation is constructed of a block number and a shuffled
+list of upkeep IDs (big.Int), json encoded to a limit of max bytes. The list
+should only include eligible upkeeps (covered in a later section). Shuffling
+is done using a randomness source derived from the config digest, epoch, and
+block number. This ensure all nodes have the same source, but the results are
+random across different OCR rounds if an observation is repeated.
+
+> the shuffling should not be necessary in the observation because we 
+> cannot trust that shuffling was done once the observation is consumed in Report
```

Comment:
We still expect good nodes to shuffle their observations in order to cycle over the upkeeps so it is needed

Reply:
This is because we're limiting the observation size?

Reply:
yes

Reply:
Right.